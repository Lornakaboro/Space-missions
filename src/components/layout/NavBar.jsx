import { NavLink } from 'react-router-dom';
import logo from '../../images/logo.png';

const NavBar = () => (
  <div className="navbar">
    <img src={logo} alt="Space Traveler's Hub logo" />
    <span className="nav-brand">Space Traveler&apos;s Hub</span>
    <div className="nav-links">
      <NavLink
        to="/rockets"
        style={({ isActive }) => ({
          textDecoration: isActive ? 'underline' : 'none',
          color: isActive ? '#00d7c6' : 'RGB(246, 141, 54)',
        })}
      >
        Rockets
      </NavLink>
      <NavLink
        to="/missions"
        style={({ isActive }) => ({
          textDecoration: isActive ? 'underline' : 'none',
          color: isActive ? '#00d7c6' : 'RGB(246, 141, 54)',
        })}
      >
        Missions
      </NavLink>
      <NavLink
        to="profile"
        style={({ isActive }) => ({
          textDecoration: isActive ? 'underline' : 'none',
          color: isActive ? '#00d7c6' : 'RGB(246, 141, 54)',
        })}
      >
        My Profile
      </NavLink>
    </div>
  </div>
);

export default NavBar;
