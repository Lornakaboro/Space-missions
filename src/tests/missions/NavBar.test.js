import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import NavBar from '../../components/layout/NavBar';

describe('NavBar', () => {
  it('NavBar component matches snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <NavBar />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
