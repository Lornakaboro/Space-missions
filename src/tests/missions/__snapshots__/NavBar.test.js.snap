// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar NavBar component matches snapshot 1`] = `
<div
  className="navbar"
>
  <img
    alt="Space Traveler's Hub logo"
    src="logo.png"
  />
  <span
    className="nav-brand"
  >
    Space Traveler's Hub
  </span>
  <div
    className="nav-links"
  >
    <a
      className=""
      href="/rockets"
      onClick={[Function]}
      style={
        Object {
          "color": "RGB(246, 141, 54)",
          "textDecoration": "none",
        }
      }
    >
      Rockets
    </a>
    <a
      className=""
      href="/missions"
      onClick={[Function]}
      style={
        Object {
          "color": "RGB(246, 141, 54)",
          "textDecoration": "none",
        }
      }
    >
      Missions
    </a>
    <a
      className=""
      href="/profile"
      onClick={[Function]}
      style={
        Object {
          "color": "RGB(246, 141, 54)",
          "textDecoration": "none",
        }
      }
    >
      My Profile
    </a>
  </div>
</div>
`;
